.github/workflows/reusable_build.yml

```diff
@@ -3,49 +3,84 @@ name: Build the application for all devices and upload the artifact
 on:
   workflow_call:
     inputs:
-      upload_artifact:
-        required: true
-        type: boolean
-      uploaded_artifact_name:
+      app_repository:
+        required: false
+        default: ${{ github.repository }}
+        type: string
+      app_branch_name:
         required: false
-        default: "application_binaries"
+        default: ${{ github.ref }}
         type: string
       relative_app_directory:
         required: false
-        default: "."
+        default: .
+        type: string
+      flags:
+        required: false
+        default: ''
+        type: string
+      app_binaries_artifact:
```

Comment:
Reading [this PR](https://github.com/LedgerHQ/app-boilerplate/pull/47/files), it was not really clear to me which workflow would generate (upload) artifact vs. which one would need (download) ones.
I know this is annoying but could it be reflected in the argument names? :)